docs/Developer Documentation/Web Engine Support/index.md

```diff
@@ -1,51 +1,100 @@
+    - The minimum supported RPC spec version
+- `minProtocolVersion`
+    - The minimum supported protocol spec version
+
 #### sdl.js
-    - the SDL JS library used to interact with SDL Core
-- Any HTML / JS files used to run the application
+sdl.js contains the SDL JS library used to interact with SDL Core.
+
+#### Other supporting javascript files
+All other HTML / JS files used to run the application.
+
+### Example Application
 
 To see an example web engine application, take a look at [the example in the javascript suite on GitHub](https://github.com/smartdevicelink/sdl_javascript_suite/tree/develop/examples/webengine/hello-sdl).
 
-Once the HMI has the app bundle on it's local file system, the HMI will use information provided by the app store, as well as information defined in the manifest file in order to call `SetAppProperties` and enter the app into Core's policy table. Then, when a user selects the newly installed application, the HMI will use information from the same two sources to launch the entrypoint HTML file in an invisible iframe. Here, the app will begin execution on the head unit and eventually call `RegisterAppInterface`. When the HMI receives an `OnAppRegistered` notification signalling that the web engine app has successfully registered, it will then send Core an `ActivateApp` request.
+### Launching a Webengine App
+
+In order for a Web Engine application to appear on the SDL app list, the HMI must notify SDL Core about newly installed Web Engine applications by sending a `BasicCommunication.SetAppProperties`. The properties sent in this request will be stored in Core's policy table and the information will stay persistent between ignition cycles. 
+
+!!! note
+Not all parameters in the `AppProperties` struct apply to Web Engine Apps. Web Engine Apps should omit `endpoint` and `authToken`.
+!!!
+
+If the users activated a Web Engine application, the HMI will use information from the manifest.js to launch the entrypoint HTML (our development HMIs' do this by creating an invisible iframe). Here, the app will begin execution on the head unit and eventually call `RegisterAppInterface`. When the HMI receives an `OnAppRegistered` notification signalling that the web engine app has successfully registered, the HMI should then send Core an `ActivateApp` request.
+
+## WebSocket Server Transport
+
+In order to support the Webengine feature, a websocket server transport was added to SDL Core. This contrasts to the websocket client transport in SDL Core that is used by Java Cloud applications. When the HMI launches a web engine application, it will provide Core's hostname and port as query parameters to the entrypoint of the web engine application. This transport supports both secure and non-secure websocket communication; which is also determined by a query parameter passed to the entrypoint html file of the web engine application.
+
+These are the accepted values for the sdl-transport-role parameter:
+- ws-server
```

Comment:
I believe there needs to be a newline above a list for the HTML generator to work